### ISSUE TITLE
kubernetes

### ISSUE THREAD

**Issue by crunchingcode** [favonia/cloudflare-ddns#898]
I saw 3 years ago you in the closed issues you added a howto on deploying this using kubernetes but I can't seem to find it anywhere on the main page. (its where it says it added it) the yaml you did write seems to be a little outdated.  I was wondering if you were going to put that back and/or create a helm for it?

**Comment by favonia** [comment]
@crunchingcode Thanks. I removed the instructions mainly because I'm not using Kubernetes and cannot keep up with its community (not just the tool) for the documentation purpose (e.g., its IPv6 situation). For example, Kubernetes gives you much more freedom in organizing the configuration files, and because I'm not using it, I don't understand it enough to "bless" any particular setup. Other reasons include that there are tools such as Kompose to convert Docker Compose configurations to Kubernetes, but not the other way around as far as I know. In general, Docker Compose configuration files, while being less powerful, are understood by more tools.

(That said, I just discovered that Kompose does not seem to understand `user:"1000:1000"` that should be converted to `RunAsUser:1000` and `RunAsGroup:1000`...)

tl;dr: my main constraint is my time and energy (after my day job). I can only afford low-maintainace instructions that will remain completely relevant and correct. The Kubernetes part unfortunately was eating up lots of my time so I gave up at some point.

**Comment by favonia** [comment]
@crunchingcode PS: I just opened a PR to support `user:"1000:1000"` in Kompose: https://github.com/kubernetes/kompose/pull/1929

**Comment by favonia** [comment]
@crunchingcode Is it okay if I point out that there's a conversion tool such as Kompose to generate usable templates? Would that solve your problem?

PS: I just opened a PR to support user:"1000:1000" in Kompose: https://github.com/kubernetes/kompose/pull/1929

**Comment by crunchingcode** [comment]
I created a helm chart/package included the security context as well.  https://github.com/crunchingcode/cloudflare-ddns/tree/main if you want to take a look. In my ansible playbook I pass it the following below. I double checked it doesn't run as root.

kubectl get pod cloudflare-ddns -o jsonpath='{.spec.containers[*].securityContext.runAsUser}' -n cloudflare-ddns
1000
        release_values:
          env:
            CF_API_TOKEN: "{{ name_of_your_ansible_vault_variable }}" #you can specifiy your api token here in "" but its unwise 
            DOMAINS: "example.com,www.example.com"
            PROXIED: "true"
            IP6_PROVIDER: "none"
          service:
            type: "ClusterIP"
            port: "80"
            targetPort: "8080"
          image:
            repository: "favonia/cloudflare-ddns"
            tag: "latest"
            pullPolicy: "IfNotPresent"

**Comment by favonia** [comment]
Hi, a few quick comments: you probably want to remove all `port` or `targetPort` because the DDNS updater does not need any open ports for initiating _incoming_ traffic.

**Comment by crunchingcode** [comment]
thats fair.  when I have time I'll repackage it.

**Comment by crunchingcode** [comment]
Done. I had to set the clusterip to none.  again new to creating helm packages lol.  I also forgot to include a .helmignore which is a must since some directories like .git potentially add megabytes to helm package which exceeds the 1meg limit for helm/kubernetes.  smh.  live and learn I guess :)

**Comment by favonia** [comment]
@crunchingcode Thanks for your intention to provide instructions. I am not sure you need a service at all---maybe a deployment is enough. However, I don't have a running Kubernetes at my hand and cannot verify this.

I guess I can add a Q&A entry to mention tools like Kompose for now.

**Comment by crunchingcode** [comment]
I've recently use kompose.  its not bad.  its not as easy imho as a helm package but I get what your saying.  I only mentioned it as my use case matches perfectly with your software. Many domains. many records.  Not to mention I really like not having to include all the record id's for each one. I just didn't have a way to easily deploy it inside my kube cluster. I figured it out tho.  necessity is the mother of invention isn't that the saying?  Cheers.

**Comment by favonia** [comment]
@crunchingcode Thank you for your comment. I wish I had time to maintain the Kubernetes instructions. Does it work without a Kubernetes service? (That is, only with a Kubernetes deployment.)

**Comment by crunchingcode** [comment]
Yup.  Once I set the clusterip: none its all I had to set.  if you look at https://github.com/crunchingcode/cloudflare-ddns/blob/main/templates/service.yaml  which is default values it passes on.  theres nothing else really set except some labels/metadata.  I dont pass anything on in my playbook either in regards to service when I install the helm package. It works great.

**Comment by favonia** [comment]
@crunchingcode Hi, I believe you _are_ using a Kubernetes service in your setup, but you probably do not need it. That is, you can probably _remove `service.yaml` completely_ and only use `deployment.yaml`. Setting `ClusterIP` to `None` is a workaround, and it is a sign that you probably do not need the Kubernetes service. Sadly I don't have a running instance to verify all of these. (Issues like this are why I don't feel very confident in providing instructions.)

**Comment by crunchingcode** [comment]
I think your right.  I'll try it out if I have time tomorrow.  Again fairly new to created helms and kubernetes.  I've learned a a huge amount just running it in my homelab so yet another thing to try and learn from lol.

Doug
